stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/WebSocketUtil.java

```diff
@@ -221,7 +241,490 @@ public static ErrorValue createErrorCause(String message, String reason, BPackag
+     * @param backOfFactor
+     * @param retryConnectorConfig retry connector config
+     */
+    private static void setBackOfFactor(float backOfFactor, RetryContext retryConnectorConfig) {
+        if (backOfFactor < 0) {
+            logger.warn("{} The backOfFactor[{}] value is set to 1.0", WebSocketConstants.STATEMENT_FOR_BACK_OF_FACTOR,
+                    backOfFactor);
+            backOfFactor = (float) 1.0;
+        }
+        retryConnectorConfig.setBackOfFactor(backOfFactor);
+    }
+
+    /**
+     * Set value of the maxInterval in the retry config.
+     * @param maxInterval
+     * @param retryConnectorConfig retry connector config
+     */
+    private static void setMaxInterval(int maxInterval, RetryContext retryConnectorConfig) {
+        if (maxInterval < 0) {
+            logger.warn("{} The maxInterval[{}] value is set to 30000", WebSocketConstants.STATEMENT_FOR_MAX_INTERVAL,
+                    maxInterval);
+            maxInterval =  30000;
+        }
+        retryConnectorConfig.setMaxInterval(maxInterval);
+    }
+
+    /**
+     * Set value of the maxAttempts in the retry config.
+     * @param maxAttempts
+     * @param retryConnectorConfig retry connector config
+     */
+    private static void setMaxAttempts(int maxAttempts, RetryContext retryConnectorConfig) {
+        if (maxAttempts < 0) {
+            logger.warn("{} The maxAttempts[{}] value is set to 0", WebSocketConstants.STATEMENT_FOR_MAX_ATTEPTS,
+                    maxAttempts);
+            maxAttempts = 0;
+        }
+        retryConnectorConfig.setMaxAttempts(maxAttempts);
+    }
+
+     /**
+     * Populate the failover config.
+     * @param clientEndpointConfig a client endpoint config
+     * @param failoverClientConnectorConfig a failover client connector config
+     * @param targetUrls target urls
+     */
+    public static void populateFailoverConnectorConfig(MapValue<String, Object> clientEndpointConfig,
+                                                       FailoverContext failoverClientConnectorConfig,
+                                                       ArrayList<String> targetUrls) {
+        int failoverInterval = Integer.parseInt(clientEndpointConfig.get(WebSocketConstants.FAILOVER_INTEVAL)
+                .toString());
+        if (failoverInterval < 0) {
+            logger.warn("{} The failoverInterval [{}] value is set to 1.0",
+                    WebSocketConstants.STATEMENT_FOR_FAILOVER_INTERVAL, failoverInterval);
+            failoverInterval = 1000;
+        }
+        failoverClientConnectorConfig.setFailoverInterval(failoverInterval);
+        failoverClientConnectorConfig.setTargetUrls(targetUrls);
+    }
+
+    /**
+     * Get the webSocket service.
+     * @param clientEndpointConfig a client endpoint config
+     * @param strand a strand
+     * @return webSocketService
+     */
+    public static WebSocketService getWebSocketService(MapValue<String, Object> clientEndpointConfig, Strand strand) {
```

Comment:
The names must reveal intent. This function seems to be creating a service rather than getting it.
Let us rename:
```suggestion
    public static WebSocketService validateAndCreateWebSocketService(MapValue<String, Object> clientEndpointConfig, Strand strand) {
```